azDevOps/azure/templates/jobs/build-dotnet.yml

```diff
@@ -74,21 +81,27 @@ jobs:
         projectKey: ${{ parameters.Sonar_projectKey }}
         projectName: ${{ parameters.Sonar_projectName }}
         projectVersion: ${{ parameters.Sonar_projectVersion }}
+        extraProperties: ${{ parameters.Sonar_extraProperties }}
       displayName: 'SonarCloud: Prepare for Analysis'
 
+# SonarCloud wants a build to analyse
     - task: DotNetCoreCLI@2
       inputs:
         command: 'build'
         workingDirectory: ${{ parameters.static_code_analysis_build_path }}
 
+# Set the code_coverage_arguments variable to make `dotnet test` output coverage
+    - bash: |
+      echo "##vso[task.setvariable variable=code_coverage_arguments;isOutput=true]/p:CollectCoverage=true /p:CoverletOutputFormat=opencover /p:CoverletOutput='./'"
```

Comment:
The reason why this isn't set at the top is because I only want to generate coverage if static_code_analysis is true 

There's no need to escape single quotes inside heredoc strings I don't believe? (The pipe | above)